client/src/components/CAHelper.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright © 2023, SAS Institute Inc., Cary, NC, USA. All Rights Reserved.
+// Licensed under SAS Code Extension Terms, available at Code_Extension_Agreement.pdf
+
+import * as fs from "fs";
+import * as https from "https";
+import * as tls from "tls";
+import { workspace } from "vscode";
+
+export const installCAs = () => {
+  const certFiles: string[] = workspace
+    .getConfiguration("SAS")
+    .get("userProvidedCertificates");
+  if (!certFiles || !certFiles.length) {
+    return;
+  }
+
+  const CAs = [].concat(tls.rootCertificates);
```

Comment:
Are we using this just to clone `rootCertificates`? If so, can we use something like this instead...
```suggestion
  const CAs = [ ...tls.rootCertificates];
```

Reply:
I revised a little. It doesn't need to copy root certificates before assigning to agent.
Are you still thinking spread operator better than `concat` in the latest code? I'm thinking `concat` should have better performance than spread operator.

Reply:
In general, `concat` seems to be more performant (for large arrays). That said, the spread syntax to my eye more clearly highlights what ~is~ was being done. Personally, I think most of the time (I know we have cases where this isn't the case), code that is easier to understand is much better than code that is performant (where the performance savings aren't significant).

All that said, I think your revised approach looks much better :)